### ISSUE TITLE
Challenges: Configurable chance to receive a reward

### ISSUE THREAD

**Issue by Talabrek** [rlf/uSkyBlock#251]
It would be nice to be able to set a percentage chance that a player will receive certain items and commands in the rewards section for challenges.

As an example:

**repeatReward:**
          **text: '2 leather'**
          **items: '334:2'**

could be changed to:

**repeatReward:**
          **text: '2 leather, 10% chance for an iron ingot'**
          **items: '334:2 265:1%10'**

and an example for a command reward:

**commands:**
            **- 'op:effect {party} 10'**

could be something like:

**commands:**
            **- '%10:op:effect {party} 10'**

which would give the command a 10% chance of executing when the challenge is completed.

**Comment by rlf** [comment]
Good suggestion. As you might have noticed - you are now a first-order collaborator on this project :) With the power to do most things (I assume).

I think we should consider splitting the `items:` section into a string-list then - those strings are getting too complex.

I.e.
```yml
repeatReward:
  text: '2 leather & chance at ingot'
  items:
   - '334:2'
   - '{p=0.1} 265:1'
  commands:
  - '{p=0.25} op:effect {party} 10'
```
I think making the "probability" part of a "more generic" way of supplying arguments, will help us "down the road" - i.e. we can easily find it (if the string starts with matching `{` and `}` - and we can easily expand it to include other "arguments/modifiers".

**Comment by Talabrek** [comment]
Looks good! Also, thanks for adding me to the project. As I become more familiar with git I will start making some direct contributions.

**Comment by dutchy1001** [comment]
Welcome back Talabrek, glad to have you onboard too

**Comment by rlf** [comment]
Just a thought - perhaps we should consider adding a "one of these randomly" as well.

I.e.
```yml
repeatReward:
  items:
  - '{p=0.1} 264:3 {p=0.2} 264:2 {p=0.7} 264:1'
  - '266:10'
  - '{p=0.01} 322:2' 
```
Would mean: `10 gold ingots, and 1-3 diamonds and a chance of a golden apple'

**Comment by dutchy1001** [comment]
Sounds good to me. 

Not sure to add this here or make a new issue, when more then 8 challenges are present ti wont go to the next row. It acceps 9 challenges per row even that there are only 8 spots left, the 9th will not show and can only be completed with chat commands. Having more then 9 challenges it will error in console after the command /c or /challenges is given. (see chat with wesley in spigot)

**Comment by rlf** [comment]
I know - post it as a separate issue. Possibly also highligt the current "limitations" to the Challenge GUI.

**Comment by rlf** [comment]
Fixed in v2.2-HF11 - at least initially.
